dataplane/apigen/ccgen/ccgen.go

```diff
@@ -416,46 +463,64 @@ var typeToUnionAccessor = map[string]*unionAccessor{
 	"sai_acl_field_data_t sai_ip4_t": {
 		accessor:        "ip4",
 		convertFromFunc: "convert_from_acl_field_data",
+		convertToFunc:   "convert_to_acl_field_data",
+		protoAccessor:   "ip",
 		aType:           acl,
 	},
 	"sai_acl_action_data_t sai_object_id_t": {
 		accessor:        "oid",
 		convertFromFunc: "convert_from_acl_action_data",
+		convertToFunc:   "convert_to_acl_action_data",
+		protoAccessor:   "oid",
 		aType:           acl,
 	},
 	"sai_acl_action_data_t sai_packet_action_t": {
 		accessor:        "s32",
 		convertFromFunc: "convert_from_acl_action_data_action",
+		convertToFunc:   "convert_to_acl_action_data_action",
+		protoAccessor:   "packet_action",
 		aType:           acl,
 	},
 	"sai_acl_field_data_t sai_acl_ip_type_t": {
 		accessor:        "s32",
 		convertFromFunc: "convert_from_acl_field_data_ip_type",
+		convertToFunc:   "convert_to_acl_field_data_ip_type",
+		protoAccessor:   "ip_type",
 		aType:           acl,
 	},
 	"sai_acl_field_data_t sai_uint8_t": {
 		accessor:        "u8",
 		convertFromFunc: "convert_from_acl_field_data",
+		convertToFunc:   "convert_to_acl_field_data_u8",
+		protoAccessor:   "uint",
```

Comment:
Both u8 and u16 are both handled by uint proto accessor? Do we need to make it more specific?

Reply:
protobuf only has 32 and 54 bit ints, so u16 and and u32 use the proto field